### PR TITLE
added TIMEOUT_MULTIPLIER support

### PR DESCRIPTION
**Pull request by daomry**
when TIMEOUT_SCALE env var is a positive value, it scales retry/steps timeouts and delays

### CODE REVIEW

File: packages/testing/src/timeouts.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { isDebugMode } from './debug-tests';
+
+const getTimeoutScale = () => {
+    const multiplierEnv = (globalThis as { process?: { env: { TIMEOUT_SCALE?: string } } })?.process?.env
+        ?.TIMEOUT_SCALE;
```
Comment (barak007) [inline packages/testing/src/timeouts.ts:5]: maybe `TIMEOUT_SCALE_FACTOR` ?

File: packages/testing/src/debug-tests.ts [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
-if (debug === 'true' || parseInt(debug || '0') > 0) {
+if (getDebug()) {
     // eslint-disable-next-line no-console
     console.log('Testing in debug mode');
-    (globalThis as { mocha?: Mocha })?.mocha?.timeout(0);
 }
 
+if (getDebug()) {
+    beforeEach('wrap mocha runnables to save ctx', function () {
+        this.currentTest?.timeout(0);
```
Comment (barak007) [inline packages/testing/src/debug-tests.ts:14]: what is this doing?

File: packages/testing/src/timeouts.ts (hunk 1 above)
Comment (daomry) [reply packages/testing/src/timeouts.ts:5]: too long IMO, how about SCALE_TIMEOUT?

File: packages/testing/src/debug-tests.ts (hunk 2 above)
Comment (daomry) [reply packages/testing/src/debug-tests.ts:14]: these tests hang when in debug mode. very annoying

File: packages/testing/src/timeouts.ts (hunk 1 above)
Comment (barak007) [reply packages/testing/src/timeouts.ts:16]: TIMEOUT_MULTIPLIER?

**Review by barak007 — APPROVED** [review]
This is wired that the debug in the test. I would remove it. but it's not a breaker for me

**Pull request merged**